src/submission/validation-clinical/surgery.ts

```diff
@@ -199,3 +188,40 @@ function checkSurgeryTypeEquality(
 
 	return existingSurgeryType === therapyRecordSurgeryType;
 }
+
+// Prevents submitting duplicate records; allows updating existing records with new values
+function checkSurgeryDuplicateOrUpdate(
+	therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+	prevSurgeries: DeepReadonly<Therapy>[],
+	existingSurgery: DeepReadonly<Therapy> | undefined,
+	errors: SubmissionValidationError[],
+) {
+	// Clone Submitted Record, minus index key, which is not stored on Clinical Records
+	const submissionClone = (({ index, ...rest }) => ({ ...rest }))(therapyRecord);
+
+	// Determine if Submission is duplicating existing record, or if Submission is an update
+	// Sort insures [A,B,C] is not compared with [A,C,B]
+	const submissionValues = Object.values(submissionClone).sort();
+	const existingRecordValues = existingSurgery
+		? Object.values(existingSurgery.clinicalInfo).sort()
+		: [];
+	const submissionIsDuplicate =
+		submissionValues.length === existingRecordValues.length &&
+		isEqual(submissionValues, existingRecordValues);
```

Comment:
Members of `submissionValues` could be arrays, these arrays will not be found as equal since they are compared by reference value. Therefore this `isEqual` will not work in cases with therapies that have array properties. This may not be the case in the current dictionary, but for future proofing we will need a recursive equal check.

Reply:
I've tested this with array values and it worked but will give this a second look

Reply:
From the Docs: "This method supports comparing arrays"
https://lodash.com/docs/4.17.15#isEqual

I added array values to the unit test for validation